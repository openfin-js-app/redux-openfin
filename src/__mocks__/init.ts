import {
    ChannelType, IConfig, IInitState,
} from "../init";

export const initState:IInitState ={
    fin:{},
    store:{
        dispatch:jest.fn(),
        getState:()=>({}),
        subscribe:jest.fn(),
        replaceReducer:jest.fn(),
    },
};