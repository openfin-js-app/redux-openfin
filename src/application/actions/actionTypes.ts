import makeType from '../../utils/makeType';

export const NEW_APPLICATION = makeType('NEW_APPLICATION');
export const NEW_APPLICATION_RES = makeType('NEW_APPLICATION_RES');

export const RESTART = makeType('APPLICATION_RESTART');
export const RESTART_RES = makeType('APPLICATION_RESTART_RES');

export const CLOSE = makeType('APPLICATION_CLOSE');
export const CLOSE_RES = makeType('APPLICATION_CLOSE_RES');
