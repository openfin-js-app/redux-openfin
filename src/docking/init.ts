import {FinWindow} from '../GlobalTypes';

import toLegacyFinWindow from '../utils/toLegacyFinWindow';

import {
    IDockingOptions, IRectangle
} from './DockingType'

import {
    getAppId, requestMonitorInfo,
    parsePositiveInt, parseOpacity,
} from './DockingUtil';

import LocalStoragePersistence from './LocalStoragePersistence';
import DockingManager from './DockingManager';


export type State = {
    monitors:IRectangle[],
    persistenceService:LocalStoragePersistence,
    options:IDockingOptions,
    dockingManager:DockingManager,
}

const DOCKING_NAMESPACE_PREFIX = 'AL90_REDUX_OPENFIN_DOCKING::';

export const initState:State = {
    monitors:[],
    persistenceService: null,
    options:null,
    dockingManager:null,
};

const defaultDockingOptions:IDockingOptions = {
    range: 20,
    spacing: 0,
    undockOffsetX: 25,
    undockOffsetY: 25,
    movingOpacity: 0.6,
    snappedMovingOpacity: 0.8,
    snappedTargetOpacity: 1,
    dockableToOthers: true,
    unregisterOnClose: true,
};

async function initMonitorInfo(fin) {
    const monitors:IRectangle[] = await requestMonitorInfo(fin);
    for (const monitorInfo of monitors) {
        // add to monitors array, rather than replacing the ref, in case window already initialised with ref
        initState.monitors.push(monitorInfo);
    }
    
}

export default (fin,mainFinWindow:FinWindow,dockingOptions:Partial<IDockingOptions>)=>{

    const options = Object.assign(defaultDockingOptions,dockingOptions);

    initMonitorInfo(fin);

    options.range = parsePositiveInt(options.range,defaultDockingOptions.range);
    options.spacing = parsePositiveInt(options.spacing,defaultDockingOptions.spacing);
    options.undockOffsetX = parsePositiveInt(options.undockOffsetX,defaultDockingOptions.undockOffsetX);
    options.undockOffsetY = parsePositiveInt(options.undockOffsetY,defaultDockingOptions.undockOffsetY);
    options.movingOpacity = parseOpacity(options.movingOpacity,defaultDockingOptions.movingOpacity);
    options.snappedMovingOpacity = parseOpacity(options.snappedMovingOpacity,defaultDockingOptions.snappedMovingOpacity);
    options.snappedTargetOpacity = parseOpacity(options.snappedTargetOpacity,defaultDockingOptions.snappedTargetOpacity);

    initState.persistenceService = new LocalStoragePersistence(DOCKING_NAMESPACE_PREFIX+getAppId(fin));

    initState.options = options;

    initState.dockingManager = new DockingManager(options);

    // register global unDockAll hot key
    fin.desktop.GlobalHotkey.register("Shift+CommandOrControl+U",()=>{
        if (initState.dockingManager){
            initState.dockingManager.undockAll();
        }
    });

    initState.dockingManager.register(toLegacyFinWindow(mainFinWindow),false)

}