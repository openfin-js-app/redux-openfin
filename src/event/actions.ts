import {EventConstant, appEvents, windowEvents} from './constants';

export type ActionDict = {
    [name:string]:EventConstant,
}

const appEventDictByName:ActionDict = {};
const appEventDictByType:ActionDict = {};
const windowEventDictByName:ActionDict = {};
const windowEventDictByType:ActionDict = {};

appEvents.forEach(oneEvent => {
    appEventDictByName[oneEvent.name] = oneEvent;
    appEventDictByType[oneEvent.type] = oneEvent;
});

windowEvents.forEach(oneEvent => {
    windowEventDictByName[oneEvent.name] = oneEvent;
    windowEventDictByType[oneEvent.type] = oneEvent;
});

export default {
    appEventDictByName,
    appEventDictByType,
    windowEventDictByName,
    windowEventDictByType,
};