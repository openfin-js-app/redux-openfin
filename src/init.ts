import { Store } from 'redux';

import registerDefaultListener from './event/registerDefaultListener';

export interface IConfig{
    ignoreStore?:boolean
}

export interface IInitState {
    fin:any,
    store:Store<any>,
}

export const initState:IInitState ={
    fin:null,
    store:null,
};

export default (fin: any, config:IConfig, store?: Store<any>)=>{

    if (store){
        registerDefaultListener(fin,store);
    }
    initState.fin = fin;

    if (config.ignoreStore){
        initState.store = null;
    }else{
        initState.store = store;
    }

}