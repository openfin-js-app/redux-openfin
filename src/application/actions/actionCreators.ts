import createFSA, {ActionCreator} from '../../utils/createFSA';
import * as types from "../types";
import * as actionTypes from "./actionTypes";

export const startApplication:ActionCreator<types.StartApplicationPayload> = createFSA<types.StartApplicationPayload>(
    actionTypes.START_APPLICATION,
    (options:types.StartApplicationPayload) => <any> options
);

export const getCurrent:ActionCreator<types.GetCurrentPayload> = createFSA<types.GetCurrentPayload>(
    actionTypes.APPLICATION_GET_CURRENT,
    (options:types.GetCurrentPayload) => <any> options
);

export const getWindow:ActionCreator<types.GetWindowPayload> = createFSA<types.GetWindowPayload>(
    actionTypes.APPLICATION_GET_WINDOW,
    (options:types.GetWindowPayload) => <any> options
);

export const wrap:ActionCreator<types.WrapPayload> = createFSA<types.WrapPayload>(
    actionTypes.APPLICATION_WRAP,
    (options:types.WrapPayload) => <any> options
);

export const addEventListener:ActionCreator<types.AddListenerPayload> = createFSA<types.AddListenerPayload>(
    actionTypes.APPLICATION_ADD_LISTENER,
    (options:types.AddListenerPayload) => <any> options
);

export const quit:ActionCreator<types.QuitPayload> = createFSA<types.QuitPayload>(
    actionTypes.APPLICATION_QUIT,
    (options:types.QuitPayload) => <any> options
);

export const getChildWindows:ActionCreator<types.GetChildWindowsPayload> = createFSA<types.GetChildWindowsPayload>(
    actionTypes.APPLICATION_GET_CHILD_WINDOWS,
    (options:types.GetChildWindowsPayload) => <any> options
);

export const getInfo:ActionCreator<types.GetInfoPayload> = createFSA<types.GetInfoPayload>(
    actionTypes.APPLICATION_GET_INFO,
    (options:types.GetInfoPayload) => <any> options
);

export const getShortcuts:ActionCreator<types.GetShortcutsPayload> = createFSA<types.GetShortcutsPayload>(
    actionTypes.APPLICATION_GET_SHORTCUTS,
    (options:types.GetShortcutsPayload) => <any> options
);

export const getTrayIconInfo:ActionCreator<types.GetTrayIconInfoPayload> = createFSA<types.GetTrayIconInfoPayload>(
    actionTypes.APPLICATION_GET_TRAY_ICON_INFO,
    (options:types.GetTrayIconInfoPayload) => <any> options
);

export const getZoomLevel:ActionCreator<types.GetZoomLevelPayload> = createFSA<types.GetZoomLevelPayload>(
    actionTypes.APPLICATION_GET_ZOOM_LEVEL,
    (options:types.GetZoomLevelPayload) => <any> options
);

export const isRunning:ActionCreator<types.IsRunningPayload> = createFSA<types.IsRunningPayload>(
    actionTypes.APPLICATION_IS_RUNNING,
    (options:types.IsRunningPayload) => <any> options
);

export const removeListener:ActionCreator<types.RemoveListenerPayload> = createFSA<types.RemoveListenerPayload>(
    actionTypes.APPLICATION_REMOVE_LISTENER,
    (options:types.RemoveListenerPayload) => <any> options
);

export const restart:ActionCreator<types.RestartPayload> = createFSA<types.RestartPayload>(
    actionTypes.RESTART,
    (options:types.RestartPayload) => <any> options
);

export const scheduleRestart:ActionCreator<types.ScheduleRestartPayload> = createFSA<types.ScheduleRestartPayload>(
    actionTypes.APPLICATION_SCHEDULE_RESTART,
    (options:types.ScheduleRestartPayload) => <any> options
);


export const setShortcuts:ActionCreator<types.SetShortcutsPayload> = createFSA<types.SetShortcutsPayload>(
    actionTypes.APPLICATION_SET_SHORTCUTS,
    (options:types.SetShortcutsPayload) => <any> options
);

export const setTrayIcon:ActionCreator<types.SetTrayIconPayload> = createFSA<types.SetTrayIconPayload>(
    actionTypes.APPLICATION_SET_TRAY_ICON,
    (options:types.SetTrayIconPayload) => <any> options
);

export const setZoomLevel:ActionCreator<types.SetZoomLevelPayload> = createFSA<types.SetZoomLevelPayload>(
    actionTypes.APPLICATION_SET_ZOOM_LEVEL,
    (options:types.SetZoomLevelPayload) => <any> options
);

export const terminate:ActionCreator<types.TerminatePayload> = createFSA<types.TerminatePayload>(
    actionTypes.APPLICATION_TERMINATE,
    (options:types.TerminatePayload) => <any> options
);