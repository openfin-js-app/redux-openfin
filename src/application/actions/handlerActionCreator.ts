import {Action} from 'redux-actions';
import createFSA, {ActionCreator} from "../../utils/createFSA";
import * as types from "../types";
import * as actionTypes from "./actionTypes";

export const startApplicationRes:ActionCreator<types.StartApplicationResPayload> = createFSA<types.StartApplicationResPayload>(
    actionTypes.START_APPLICATION_RES,
    (options:types.StartApplicationResPayload) => <any> options
);

export const getCurrentRes:ActionCreator<types.GetCurrentResPayload> = createFSA<types.GetCurrentResPayload>(
    actionTypes.APPLICATION_GET_CURRENT_RES,
    (options:types.GetCurrentResPayload) => <any> options
);

export const getWindowRes:ActionCreator<types.GetWindowResPayload> = createFSA<types.GetWindowResPayload>(
    actionTypes.APPLICATION_GET_WINDOW_RES,
    (options:types.GetWindowResPayload) => <any> options
);

export const wrapRes:ActionCreator<types.WrapResPayload> = createFSA<types.WrapResPayload>(
    actionTypes.APPLICATION_WRAP_RES,
    (options:types.WrapResPayload) => <any> options
);

export const addListenerRes:ActionCreator<types.AddListenerResPayload> = createFSA<types.AddListenerResPayload>(
    actionTypes.APPLICATION_ADD_LISTENER_RES,
    (options:types.AddListenerResPayload) => <any> options
);

export const quitRes:ActionCreator<types.QuitResPayload> = createFSA<types.QuitResPayload>(
    actionTypes.APPLICATION_QUIT_RES,
    (options:types.QuitResPayload) => <any> options
);

export const getChildWindowsRes:ActionCreator<types.GetChildWindowsResPayload> = createFSA<types.GetChildWindowsResPayload>(
    actionTypes.APPLICATION_GET_CHILD_WINDOWS_RES,
    (options:types.GetChildWindowsResPayload) => <any> options
);

export const getInfoRes:ActionCreator<types.GetInfoResPayload> = createFSA<types.GetInfoResPayload>(
    actionTypes.APPLICATION_GET_INFO_RES,
    (options:types.GetInfoResPayload) => <any> options
);

export const getShortcutsRes:ActionCreator<types.GetShortcutsResPayload> = createFSA<types.GetShortcutsResPayload>(
    actionTypes.APPLICATION_GET_SHORTCUTS_RES,
    (options:types.GetShortcutsResPayload) => <any> options
);

export const getTrayIconInfoRes:ActionCreator<types.GetTrayIconInfoResPayload> = createFSA<types.GetTrayIconInfoResPayload>(
    actionTypes.APPLICATION_GET_TRAY_ICON_INFO_RES,
    (options:types.GetTrayIconInfoResPayload) => <any> options
);

export const getZoomLevelRes:ActionCreator<types.GetZoomLevelResPayload> = createFSA<types.GetZoomLevelResPayload>(
    actionTypes.APPLICATION_GET_ZOOM_LEVEL_RES,
    (options:types.GetZoomLevelResPayload) => <any> options
);

export const isRunningRes:ActionCreator<types.IsRunningResPayload> = createFSA<types.IsRunningResPayload>(
    actionTypes.APPLICATION_IS_RUNNING_RES,
    (options:types.IsRunningResPayload) => <any> options
);

export const removeListenerRes:ActionCreator<types.RemoveListenerResPayload> = createFSA<types.RemoveListenerResPayload>(
    actionTypes.APPLICATION_REMOVE_LISTENER_RES,
    (options:types.RemoveListenerResPayload) => <any> options
);

export const restartRes:ActionCreator<types.RestartResPayload> = createFSA<types.RestartResPayload>(
    actionTypes.RESTART_RES,
    (options:types.RestartResPayload) => <any> options
);

export const scheduleRestartRes:ActionCreator<types.ScheduleRestartResPayload> = createFSA<types.ScheduleRestartResPayload>(
    actionTypes.APPLICATION_SCHEDULE_RESTART_RES,
    (options:types.ScheduleRestartResPayload) => <any> options
);


export const setShortcutsRes:ActionCreator<types.SetShortcutsResPayload> = createFSA<types.SetShortcutsResPayload>(
    actionTypes.APPLICATION_SET_SHORTCUTS_RES,
    (options:types.SetShortcutsResPayload) => <any> options
);

export const setTrayIconRes:ActionCreator<types.SetTrayIconResPayload> = createFSA<types.SetTrayIconResPayload>(
    actionTypes.APPLICATION_SET_TRAY_ICON_RES,
    (options:types.SetTrayIconResPayload) => <any> options
);

export const setZoomLevelRes:ActionCreator<types.SetZoomLevelResPayload> = createFSA<types.SetZoomLevelResPayload>(
    actionTypes.APPLICATION_SET_ZOOM_LEVEL_RES,
    (options:types.SetZoomLevelResPayload) => <any> options
);

export const terminateRes:ActionCreator<types.TerminateResPayload> = createFSA<types.TerminateResPayload>(
    actionTypes.APPLICATION_TERMINATE_RES,
    (options:types.TerminateResPayload) => <any> options
);
