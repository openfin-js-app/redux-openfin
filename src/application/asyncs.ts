import {Action} from 'redux-actions';
import * as types from "./types";
import * as handlerActions from './actions/handlerActionCreator';

import createAsyncFun from '../utils/createAsyncFun'

import {
    NEW_APPLICAITON_ERROR_MSG,
    RESTART_ERROR_MSG,
    CLOSE_ERROR_MSG,
} from './types';

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#Application
export async function newApplication(action:Action<types.NewApplicationPayload>):Promise<Action<types.NewApplicationResPayload>>{
    const  options  = action.payload;
    return createAsyncFun<types.NewApplicationPayload,types.NewApplicationResPayload>(
        action,
        NEW_APPLICAITON_ERROR_MSG,
        handlerActions.newApplicationRes,
        (fin,action,resActionCreator,succCb,errCb)=>{
            let app = new fin.desktop.Application(options,
                (app:any)=>{
                    const responseAction = resActionCreator({app});
                    succCb(responseAction);
                },errCb);
        }
    );
}

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#.getCurrent

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#getWindow

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#.wrap

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#addEventListener

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#close
export async function close(action:Action<types.ClosePayload>):Promise<Action<types.CloseResPayload>>{
    const { force } = action.payload;
    return createAsyncFun<types.ClosePayload,types.CloseResPayload>(
        action,
        CLOSE_ERROR_MSG,
        handlerActions.closeRes,
        (fin,action,resActionCreator,succCb,errCb)=>{
            let application = fin.desktop.Application.getCurrent();
            application.close(force,
                ()=>{
                    const responseAction = resActionCreator({});
                    succCb(responseAction);
                },errCb);
        }
    );
}

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#getChildWindows

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#getInfo

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#getShortcuts

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#getTrayIconInfo

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#getZoomLevel

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#isRunning

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#removeEventListener

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#restart
export async function restart(action:Action<types.RestartPayload>):Promise<Action<types.RestartResPayload>>{
    return createAsyncFun<types.RestartPayload,types.NewApplicationResPayload>(
        action,
        RESTART_ERROR_MSG,
        handlerActions.restartRes,
        (fin,action,resActionCreator,succCb,errCb)=>{
            let application = fin.desktop.Application.getCurrent();
            application.restart(
                ()=>{
                    const responseAction = resActionCreator({});
                    succCb(responseAction);
                },errCb);
        }
    );
}

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#run

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#scheduleRestart

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#setShortcuts

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#setTrayIcon

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#setZoomLevel

//http://cdn.openfin.co/jsdocs/beta/fin.desktop.Application.html#terminate
