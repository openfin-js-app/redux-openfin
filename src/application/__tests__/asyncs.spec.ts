jest.mock('../../utils/createAsyncFun');
const createAsyncFun = require('../../utils/createAsyncFun');

import {initState} from '../../init';
import * as actions from '../actions';
import * as asyncs  from '../asyncs';

describe('Application asyncs',()=>{

    afterEach(()=>{
        jest.resetAllMocks();
    });

    it('newApplication',async ()=>{
        const fin = {
            desktop:{
                Application : jest.fn((options,succCb,errCb)=>{
                    succCb();
                })
            }
        };
        const succCb = jest.fn();
        const errCb = jest.fn();

        createAsyncFun.default=jest.fn(
            (action,ERROR_MSG,resActionCreator,finCb)=>{
                finCb(fin,action,resActionCreator,succCb,errCb);
            }
        );
        await asyncs.newApplication(
            actions.newApplication({
                uuid:'uuid',
                name:'name',
                mainWindowOptions:{},
            })
        );
        expect(fin.desktop.Application).toHaveBeenCalled();
        expect(succCb).toHaveBeenCalled();
        expect(createAsyncFun).toMatchSnapshot();
    });

    it('restart',async ()=>{
        const restart = jest.fn((succCb,errCb)=>{
            succCb();
        });
        initState.currentApplication = {
            restart
        } as any;
        const succCb = jest.fn();
        const errCb = jest.fn();
        createAsyncFun.default=jest.fn(
            (action,ERROR_MSG,resActionCreator,finCb)=>{
                finCb({},action,resActionCreator,succCb,errCb);
            }
        );
        await asyncs.restart(
            actions.restart({})
        );
        expect(restart).toMatchSnapshot();
        expect(succCb).toMatchSnapshot();
        expect(createAsyncFun).toMatchSnapshot();
    });

    it('close',async ()=>{
        const close = jest.fn((force,succCb,errCb)=>{
            succCb();
        });
        const fin = {
            desktop:{
                Application:{
                    getCurrent:()=>{
                        return {
                            close
                        }
                    }
                }
            }
        };
        const succCb = jest.fn();
        const errCb = jest.fn();
        createAsyncFun.default=jest.fn(
            (action,ERROR_MSG,resActionCreator,finCb)=>{
                finCb(fin,action,resActionCreator,succCb,errCb);
            }
        );

        await asyncs.close(
            actions.close({force:true})
        );
        expect(createAsyncFun).toMatchSnapshot();
    });

});