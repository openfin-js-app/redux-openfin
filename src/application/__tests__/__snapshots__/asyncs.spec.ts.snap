// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application asyncs close 1`] = `
Object {
  "__esModule": true,
  "default": [MockFunction] {
    "calls": Array [
      Array [
        Object {
          "error": undefined,
          "payload": Object {
            "force": true,
          },
          "type": "REDUX_OPENFIN_REQ::APPLICATION_CLOSE",
        },
        "OpenFin API call Application.close() failed.",
        [Function],
        [Function],
      ],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": undefined,
      },
    ],
  },
}
`;

exports[`Application asyncs newApplication 1`] = `
Object {
  "__esModule": true,
  "default": [MockFunction] {
    "calls": Array [
      Array [
        Object {
          "error": undefined,
          "payload": Object {
            "mainWindowOptions": Object {},
            "name": "name",
            "uuid": "uuid",
          },
          "type": "REDUX_OPENFIN_REQ::NEW_APPLICATION",
        },
        "OpenFin API call Application.constructor failed.",
        [Function],
        [Function],
      ],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": undefined,
      },
    ],
  },
}
`;

exports[`Application asyncs restart 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      [Function],
      [MockFunction],
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`Application asyncs restart 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "error": undefined,
        "payload": Object {},
        "type": "REDUX_OPENFIN_RES::APPLICATION_RESTART_RES",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`Application asyncs restart 3`] = `
Object {
  "__esModule": true,
  "default": [MockFunction] {
    "calls": Array [
      Array [
        Object {
          "error": undefined,
          "payload": Object {},
          "type": "REDUX_OPENFIN_REQ::APPLICATION_RESTART",
        },
        "OpenFin API call Application.restart() failed.",
        [Function],
        [Function],
      ],
    ],
    "results": Array [
      Object {
        "isThrow": false,
        "value": undefined,
      },
    ],
  },
}
`;
